src/components/sidebar/CallTreeSidebar.js

```diff
@@ -43,26 +98,44 @@ class CallTreeSidebar extends React.PureComponent<StateProps> {
       );
     }
 
-    const data = tree.getDisplayData(selectedNodeIndex);
-    // `data.selfTime` is a string, containing either a number or, if the value
-    // is 0, is '—'. So we we use isNaN on purpose (instead of Number.isNaN), to
-    // force a conversion and decide whether we should add the unit or keep the
-    // character '—'.
-    // We don't compare against '—' to avoid hardcoded values. In the future we
-    // should have a dedicated method in `tree` to recover the values we need in
-    // the format we need.
-    const selfTime = isNaN(data.selfTime)
-      ? data.selfTime
-      : data.selfTime + 'ms';
+    const totalTimePercent = Math.round(totalTime / rootTime * 100);
+    const selfTimePercent = Math.round(selfTime / rootTime * 100);
+    const totalTimeForFuncPercent = Math.round(
+      totalTimeForFunc / rootTime * 100
+    );
+    const selfTimeForFuncPercent = Math.round(selfTimeForFunc / rootTime * 100);
+
     return (
       <aside className="sidebar sidebar-calltree">
         <header className="sidebar-titlegroup">
-          <h2 className="sidebar-title">{getFunctionName(data.name)}</h2>
-          <p className="sidebar-subtitle">{data.lib}</p>
+          <h2 className="sidebar-title">{name}</h2>
+          {lib ? <p className="sidebar-subtitle">{lib}</p> : null}
         </header>
+        <h3 className="sidebar-title2">About the selected path</h3>
         <div className="sidebar-details">
-          <SidebarDetail label="Running Time">{data.totalTime}ms</SidebarDetail>
-          <SidebarDetail label="Self Time">{selfTime}</SidebarDetail>
+          <SidebarDetail label="Running Time">
+            {totalTime}ms ({totalTimePercent}%)
+          </SidebarDetail>
+          <SidebarDetail label="Self Time">
+            {selfTime ? `${selfTime}ms (${selfTimePercent}%)` : '—'}
+          </SidebarDetail>
+        </div>
+        {jsEngineInformations ? (
+          <React.Fragment>
+            <h4 className="sidebar-title3">JavaScript engine breakdown</h4>
+            <JsModes jsModes={jsEngineInformations} />
+          </React.Fragment>
+        ) : null}
+        <h3 className="sidebar-title2">This function in the tree</h3>
```

Comment:
The functionName can be quite long sometimes, like `mozilla::detail::RunnableMethodImpl::Run`, so that's why I didn't want to repeat it. But I'm not opposed...